components/supervisor/frontend/src/experiments/client.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import * as configcat from "configcat-js";
+import { ConfigCatClient } from "@gitpod/gitpod-protocol/lib/experiments/configcat";
+import { Client } from "@gitpod/gitpod-protocol/lib/experiments/types";
+import { LogLevel } from "configcat-common";
+import { serverUrl } from "../shared/urls";
+
+let client: Client | undefined;
+
+export function getExperimentsClient(): Client {
+    // We have already instantiated a client, we can just re-use it.
+    if (client !== undefined) {
+        return client;
+    }
+
+    client = newProxyConfigCatClient();
+    return client;
+}
+
+function newProxyConfigCatClient(): Client {
```

Comment:
Maybe we could make service.tsx pass the passed `redirectUrl` then don't need to add configcat lib to supervisor frontend?

Because `info` is updated by `service.tsx` and so that supervisor frontend know that workspace is going to be stopped.